network/nodenetwork/snapshot.go

```diff
@@ -17,9 +17,69 @@
 
 package nodenetwork
 
+import (
+	"github.com/insolar/insolar/core"
+)
+
+type ListType int
+
+const (
+	ListWorking ListType = iota
+	ListIdle
+	ListLeaving
+	ListSuspected
+	ListJoiner
+
+	ListLength
+)
+
 type Snapshot struct {
+	pulse core.PulseNumber
+	state core.NetworkState
+
+	nodeList [ListLength][]core.Node
+}
+
+func (s *Snapshot) GetPulse() core.PulseNumber {
+	return s.pulse
+}
+
+// NewSnapshot create new snapshot for pulse.
+func NewSnapshot(number core.PulseNumber, nodes map[core.RecordRef]core.Node) *Snapshot {
+	return &Snapshot{
+		pulse: number,
+		// TODO: pass actual state
+		state:    core.NoNetworkState,
+		nodeList: sortNodes(nodes),
+	}
+}
+
+// sortNodes temporary method to create snapshot lists. Will be replaced by special function that will take in count
+// previous snapshot and approved claims.
+func sortNodes(nodes map[core.RecordRef]core.Node) [ListLength][]core.Node {
```

Comment:
may be "assort" or "separate" is better word for method name
There is no sorting in classic meaning here

Reply:
renamed to "splitNodes"